### PR TITLE
Osx travis

### PR DESCRIPTION
**Pull request by sergiusens**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

**Comment by kyrofa** [comment]
OSX tests seem to be failing with OsRelease errors.

File: tools/travis/run_tests.sh [hunk 1]
```diff
@@ -20,48 +20,67 @@
...
 
-if [ "$test" = "static" ]; then
+if [ "$TRAVIS_OS_NAME" = "osx" ] && [ -z "$SNAPCRAFT_FROM_BREW" ]; then
+    echo "osx tests require snapcraft from brew to be setup"
+    exit 0
+fi
+
+if [ "$TRAVIS_OS_NAME" = "osx" ]; then
+    brew upgrade python
+    # readlink -f doesn't work on osx.
```
Comment (kalikiana) [inline tools/travis/run_tests.sh]: I'm guessing this comment is misplaced here  and belongs further down where greadlink is being used?

File: tools/travis/run_tests.sh [hunk 2]
```diff
@@ -20,48 +20,67 @@
...
+    brew upgrade coreutils || brew install coreutils
```
Comment (kalikiana) [inline tools/travis/run_tests.sh]: Why the `||`? Can you add a comment what we want to achieve here?

**Comment by sergiusens** [comment]
ok, I am going ahead and merging this, any improvement to this part of the testing infrastructure is going to be an ongoing process anyways.

**Pull request merged**